File: tests/rosetta/transpiler/php/associative-array-merging.php
<?php
function merge($base, $update) {
  global $main;
  $result = [];
  foreach (array_keys($base) as $k) {
  $result[$k] = $base[$k];
};
  foreach (array_keys($update) as $k) {
  $result[$k] = $update[$k];
};
  return $result;
}
function main() {
  global $merge;
  $base = ["name" => "Rocket Skates", "price" => 12.75, "color" => "yellow"];
  $update = ["price" => 15.25, "color" => "red", "year" => 1974];
  $result = merge($base, $update);
  echo str_replace("false", "False", str_replace("true", "True", str_replace("\"", "'", str_replace(":", ": ", str_replace(",", ", ", json_encode($result, 1344)))))), PHP_EOL;
}
main();

File: tests/rosetta/transpiler/FS/associative-array-merging.fs
// Generated 2025-07-24 18:50 +0000

exception Return

let rec merge (``base``: Map<string, obj>) (update: Map<string, obj>) =
    let mutable __ret : Map<string, obj> = Unchecked.defaultof<Map<string, obj>>
    let mutable ``base`` = ``base``
    let mutable update = update
    try
        let mutable result: Map<string, obj> = Map.ofList []
        for KeyValue(k, _) in ``base`` do
            result <- Map.add k ((defaultArg (Map.tryFind k ``base``) Unchecked.defaultof<obj>)) result
        for KeyValue(k, _) in update do
            result <- Map.add k ((defaultArg (Map.tryFind k update) Unchecked.defaultof<obj>)) result
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
and main () =
    let mutable __ret : unit = Unchecked.defaultof<unit>
    try
        let ``base``: Map<string, obj> = Map.ofList [("name", box "Rocket Skates"); ("price", box 12.75); ("color", box "yellow")]
        let update: Map<string, obj> = Map.ofList [("price", box 15.25); ("color", box "red"); ("year", box 1974)]
        let result = merge ``base`` update
        printfn "%A" result
        __ret
    with
        | Return -> __ret
main()
